### ISSUE TITLE
Finite Population Corrections for `sem`

### ISSUE THREAD

**Issue by ParadaCarleton** [JuliaStats/StatsBase.jl#765]
When sampling from a finite population without replacement, the standard error is less than the current `sem` -- for instance, the standard error is 0 when you've sampled the whole population. It'd be nice to include some way to calculate the finite-population standard error.

**Comment by nalimilan** [comment]
Yes, we need a more general effort to support all kinds of survey statistics. This should probably go to a separate package, at least initially.
